framework/src/araui/widgets/ui_text_widget.c

```diff
@@ -539,13 +543,26 @@ static void _ui_text_widget_draw_func(ui_widget_t widget, uint32_t dt)
 #if defined(CONFIG_UI_ENABLE_EMOJI)
 			// If the code is emoji
 			if (is_emoji(body->utf_code[draw_idx])) {
-				if (emoji_get_asset(body->utf_code[draw_idx])) {
-					// todo
-					/*
-					ui_hal_draw_cropped_resized_image(
-						x, y, emoji_get_asset(body->utf_code[draw_idx]),
-						emoji_size, body->font_size, body->font_size);
-					*/
+				emoji_bitmap = emoji_get_bitmap(body->utf_code[draw_idx]);
```

Comment:
If I were you I would follow below.
1. Create emoji files
2. Maintain emoji file list
3. Provide API to create bitmap with file path. This will be useful for user in future.
Of course choice is yours, also it is not related to this PR.